### PR TITLE
test(stackable-versioned): Add snapshot testing

### PR DESCRIPTION
**Pull request by Techassi**
This PR adds snapshot tests to enable macro output testing.

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] (Integration-)Test cases added
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

### CODE REVIEW

File: crates/stackable-versioned-macros/fixtures/README.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    )]
+    bar: usize,
+    baz: bool,
+}
+```
+
+## Recommended Workflow
+
+First, add new input files (which automatically get picked up by `insta`) to the `fixtures/inputs`
+folder. Make sure the delimiter is placed correctly between the attribute and the container
```
Comment (NickLarsenNZ) [inline crates/stackable-versioned-macros/fixtures/README.md:41]: Probably no need to add it, but this includes doc-comments.

File: crates/stackable-versioned-macros/fixtures/README.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+definition. Next, generate the snapshot files (initially not accepted) by running
+
+```shell
+cargo insta test -p stackable-versioned-macros
+```
+
+This command will place the new snapshot files in the `fixtures/snapshots` folder. To review them,
```
Comment (NickLarsenNZ) [inline crates/stackable-versioned-macros/fixtures/README.md:48]: Could be worth mentioning that they are suffixed with `.new`, and should not be committed unless internationally (eg: for collaboration). They should not appear on the `main` branch.

File: crates/stackable-versioned-macros/fixtures/README.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+run the `cargo insta review` command. Now check these files in.
```
Comment (NickLarsenNZ) [inline crates/stackable-versioned-macros/fixtures/README.md:49]: I thought maybe some elaboration was needed:

```suggestion
run the `cargo insta review` command, then accept or fix the snapshots. Once all are accepted (ie: no `.new` files remaining), check in the files.
```

File: crates/stackable-versioned-macros/tests/trybuild.rs [hunk 4]
```diff
@@ -37,7 +26,6 @@ mod default {
 #[test]
 fn default_macros() {
     let t = trybuild::TestCases::new();
-    t.pass("tests/default/pass/*.rs");
```
Comment (NickLarsenNZ) [inline crates/stackable-versioned-macros/tests/trybuild.rs:40]: I'm assuming insta can't snapshots test the failed compilation?

Just checking if trybuild is still useful.

**Review by NickLarsenNZ — APPROVED** [review]
LGTM, a couple of suggestions and a question.

File: crates/stackable-versioned-macros/tests/trybuild.rs (hunk 4 above)
Comment (Techassi) [reply crates/stackable-versioned-macros/tests/trybuild.rs:40]: I haven't looked into it, but I have a feeling `insta` can't do it. So `trybuild` would still be needed.

File: crates/stackable-versioned-macros/fixtures/README.md (hunk 2 above)
Comment (Techassi) [reply crates/stackable-versioned-macros/fixtures/README.md]: Added in cf1ec21

File: crates/stackable-versioned-macros/fixtures/README.md (hunk 1 above)
Comment (Techassi) [reply crates/stackable-versioned-macros/fixtures/README.md:41]: Added in cf1ec21

File: crates/stackable-versioned-macros/tests/trybuild.rs (hunk 4 above)
Comment (Techassi) [reply crates/stackable-versioned-macros/tests/trybuild.rs:40]: `insta` currently cannot do it. For now, we keep the (failing) `trybuild` test cases.

**Review by NickLarsenNZ — APPROVED** [review]
I think it is still a good idea to keep the fail folder to make it obvious to the casual browser (without seeing the logic in trybuild.rs).

**Review by NickLarsenNZ — APPROVED** [review]
LGTM

**Pull request merged**